tawazi/dag.py

```diff
@@ -301,20 +322,27 @@ def get_num_running_threads(_futures: Dict[Hashable, "Future[Any]"]) -> int:
                         graph.remove_node(id_)
 
                 # 2. list the root nodes that aren't being executed
-                runnable_nodes = list(set(graph.root_nodes()) - set(futures.keys()))
+                runnable_nodes_ids = list(set(graph.root_nodes()) - set(futures.keys()))
 
                 # 3. if no runnable node exist, go to step 6 (wait for a node to finish)
                 #   (This **might** create a new root node)
-                if len(runnable_nodes) == 0:
+                if len(runnable_nodes_ids) == 0:
                     self.logger.debug("No runnable Nodes available")
                     continue
 
                 # 4. choose a node to run
-                # 4.1 get the most prioritized runnable node
-                node_id = sorted(runnable_nodes, key=lambda n: node_dict[n].priority)[-1]
+                # 4.1 get the most prioritized node to run
+                # 4.1.1 get all the nodes that have the highest priority
+                runnable_nodes = [node_dict[node_id] for node_id in runnable_nodes_ids]
+                highest_priority_nodes = get_highest_priority_nodes(runnable_nodes)
 
-                exec_node = node_dict[node_id]
-                self.logger.info(f"{node_id} will run!")
+                # 4.1.2 get the node with the highest compound priority
+                # (randomly selected if multiple are suggested)
+                exec_node = sorted(highest_priority_nodes, key=lambda node: node.compound_priority)[  # type: ignore
+                    -1
+                ]
+
+                self.logger.info(f"{exec_node.id} will run!")
```

Comment:
I thought that a user will spend more time using an INFO level of debugging. I think that it would be acceptable/useful for the user to see the next node to be executed all the time... the other messages are really internal to the working of tawazi, so I don't want to disturb the user with them... this is why they are basically only usefull for the developpers of the library (us: you and me)

Reply:
yeah but in that case don't you think that it'll pollute the logs ? With the future config to come, it might be a good idea to set the log level there anyway, and maybe we could review all the logging and see which is really useful, and which is useful for debugging